### PR TITLE
Refactor Revit test base classes.

### PR DESCRIPTION
**Pull request by ikeough**
This PR refactors the Revit base classes as follows:
- Merges the SystemTest class into RevitSystemTestBase.
- Changes all former classes which inherited from SystemTest to inherit from RevitSystemTestBase.
- Adds logic to set the DocumentManager's UIApplication and UIDocument. This logic was previously set inside RTF, but all Dynamo dependencies have been removed from RTF.

PTAL:
- [ ] @Randy-Ma 

Requires Merge:
- [ ] Revit_2015
- [ ] Revit_2016

### CODE REVIEW

File: test/Libraries/RevitTestServices/RevitSystemTestBase.cs [hunk 1]
```diff
@@ -59,7 +162,46 @@ public override void TearDown()
             TransactionManager.Instance.ForceCloseTransaction();
         }
 
-        public override void StartDynamo()
+        #endregion
+
+        #region protected methods
+
+        protected override void SetupCore()
```
Comment (Randy-Ma) [inline test/Libraries/RevitTestServices/RevitSystemTestBase.cs]: Shall we keep the access modifiers of SetupCore?

File: test/Libraries/RevitTestServices/RevitSystemTestBase.cs [hunk 2]
```diff
@@ -59,7 +162,46 @@ public override void TearDown()
...
+                emptyModelPath1 = Path.Combine(workingDirectory, "emptyV1.rfa");
+            }
+            else
+            {
+                emptyModelPath = Path.Combine(workingDirectory, "empty.rfa");
+                emptyModelPath1 = Path.Combine(workingDirectory, "empty1.rfa");
+            }
+        }
+
+        protected override void StartDynamo()
```
Comment (Randy-Ma) [inline test/Libraries/RevitTestServices/RevitSystemTestBase.cs]: Shall we keep the access modifiers of StartDynamo?

File: test/Libraries/RevitTestServices/RevitSystemTestBase.cs (hunk 1 above)
Comment (Randy-Ma) [inline test/Libraries/RevitTestServices/RevitSystemTestBase.cs]: Sorry, I see you have changed the modifier of the method in the base class.

File: test/Libraries/RevitTestServices/RevitSystemTestBase.cs (hunk 2 above)
Comment (Randy-Ma) [inline test/Libraries/RevitTestServices/RevitSystemTestBase.cs]: Ignore this comment

File: test/Libraries/RevitTestServices/RevitSystemTestBase.cs [hunk 3]
```diff
@@ -36,17 +142,14 @@ public void Shutdown()
...
+        private string defsPath;
+        protected string emptyModelPath1;
+        protected string emptyModelPath;
 
-        [SetUp]
-        public override void Setup()
-        {
-            base.Setup();
-
-            DocumentManager.Instance.CurrentUIApplication.ViewActivating += CurrentUIApplication_ViewActivating;
```
Comment (Randy-Ma) [inline test/Libraries/RevitTestServices/RevitSystemTestBase.cs]: @ikeough 
Is this required to fix something? I have an impression ViewActivating event is used for multiple test cases from a test source.

Comment (Randy-Ma) [inline test/Libraries/RevitTestServices/RevitSystemTestBase.cs]: I mean those test cases in the "RegressionTest" category in RevitSystemTests.

Comment (ikeough) [inline test/Libraries/RevitTestServices/RevitSystemTestBase.cs]: I didn't remove this hook completely. I just moved it into SetupCore. See below. Once I moved the ViewActivating hook, the Setup override was no longer needed.

**Comment by Randy-Ma** [comment]
@ikeough Thanks for the fix! Except for the one point about the ViewActivating event, LGTM! I am merging this so that at least most test cases pass (if I am wrong with the event, then hopefully all impacted test cases will pass again). As the access modifiers of two methods have changed, this need to be merged so that there will be no issues with the build.

File: test/Libraries/RevitTestServices/RevitSystemTestBase.cs (hunk 3 above)
Comment (ikeough) [inline test/Libraries/RevitTestServices/RevitSystemTestBase.cs]: @Randy-Ma I see what you mean now :( It needs to call the Setup method before each regression test. Nice catch!

**Pull request merged**